src/librustc_resolve/lib.rs

```diff
@@ -953,6 +981,19 @@ impl<'a> ModuleS<'a> {
         assert!(self.pub_glob_count.get() > 0);
         self.pub_glob_count.set(self.pub_glob_count.get() - 1);
     }
+
+    fn is_successor_of(&self, ancestor: Module<'a>) -> bool {
```

Comment:
nit: I think `is_descendant_of` would be more clear -- `successor` evokes the natural numbers and linked lists for me, not trees.